synthesizer/src/vm/finalize.rs

```diff
@@ -791,17 +792,66 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
+                            );
+                        } else {
+                            ensure!(
+                                *amount >= MIN_DELEGATOR_STAKE,
+                                "Ratify::Genesis(..) the delegator {address} must stake at least {MIN_DELEGATOR_STAKE}",
+                            );
+                            // If the address is a delegator, check that the corresponding validator is open.
+                            ensure!(
+                                committee.is_committee_member_open(*validator_address),
+                                "Ratify::Genesis(..) the delegator {address} is delegating to a closed validator {validator_address}",
+                            );
+                        }
+                        // Ensure that each staker has an entry in `public_balances`.
+                        ensure!(
+                            public_balances.contains_key(address),
+                            "Ratify::Genesis(..) the staker {address} is missing from the public balances",
+                        );
+                        // Accumulate the staked amount per validator.
+                        let total = stake_per_validator.entry(validator_address).or_insert(0u64);
+                        *total = total.saturating_add(*amount);
+                    }
+                    // Ensure the stake per validator matches the committee.
+                    ensure!(
+                        stake_per_validator.len() == committee.members().len(),
+                        "Ratify::Genesis(..) the number of validators in the committee does not match the number of validators in the bonded balances",
+                    );
+
+                    // Check that `committee` is consistent with `stake_per_validator`.
+                    for (validator_address, amount) in stake_per_validator {
+                        // Retrieve the expected validator stake from the committee.
+                        let expected_amount = match committee.members().get(validator_address) {
+                            Some((amount, _)) => *amount,
+                            None => bail!(
+                                "Ratify::Genesis(..) found a validator in the bonded balances that is not in the committee"
+                            ),
+                        };
+                        // Ensure the staked amount matches the committee.
+                        ensure!(
+                            expected_amount == amount,
+                            "Ratify::Genesis(..) inconsistent staked amount for validator {validator_address}",
+                        );
                     }
 
                     // Construct the next committee map and next bonded map.
                     let (next_committee_map, next_bonded_map) =
-                        to_next_commitee_map_and_bonded_map(committee, &stakers);
+                        to_next_commitee_map_and_bonded_map(committee, bonded_balances);
```

Comment:
Why is this `bonded_balances`?!

Reply:
How is there not a test catching this? Please share a commit that demonstrates catching this erroneous case.

Reply:
`bonded_balances` contains the expected state of the `bonded` mapping in `credits.aleo`.
The purpose of `stake_per_validator` is to check that the stakes in `committee` in fact equals the sum of the stakes in `bonded_balances`.
The commit can be found [here](https://github.com/AleoHQ/snarkVM/pull/2308/commits/33c4af3488e7ed4736ed8a4d2432ac6e4decaaf4)